src/core_plugins/kibana/public/dashboard/top_nav/show_add_panel.js

```diff
@@ -0,0 +1,44 @@
+import { DashboardAddPanel } from './add_panel';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+let isOpen = false;
+
+export function showAddPanel(savedObjectsClient, addNewPanel, addNewVis) {
+  if (isOpen) {
+    return;
+  }
+
+  isOpen = true;
+  const container = document.createElement('div');
+  const onClose = () => {
+    ReactDOM.unmountComponentAtNode(container);
+    document.body.removeChild(container);
+    isOpen = false;
+  };
+  const find = (type, search, page, perPage) => {
+    return savedObjectsClient.find({
```

Comment:
Should the logic exist in show_add_panel or saved_object_finder?

Reply:
Since the saved object finder takes in type and find, I think it kinda fits best in show_add_panel.  Then we don't have to add more `(if type == visualize { do this} )` logic inside of the saved object finder which tries to be more generic.

Eventually this add panel stuff will change too so it fits with the more generic embeddable concept, maybe once we get closer to k7 and we have a design that doesn't include the two tabs.  Then embeddables can have a generic isLab parameter and we don't need the specific type checks.